web/src/features/charts/OriginChart.tsx

```diff
@@ -72,6 +73,11 @@ function OriginChart({ displayByEmissions, datetimes, timeRange }: OriginChartPr
   const isHourly = useAtomValue(isHourlyAtom);
   const selectedData = useSelectedData(displayByEmissions);
 
+  // Call hooks before any conditional returns
+  const estimationMethod = data?.chartData?.find((d) => d.meta?.estimationMethod)?.meta
+    ?.estimationMethod;
```

Comment:
Duplicate code should be avoided otherwise our maintenance cost will go up long term.